### PR TITLE
(maint) Prep for 3.1 release

### PR DESCRIPTION
**Pull request by justinstoller**

### CODE REVIEW

**Comment by justinstoller** [comment]
@Magisus @binford2k @raphink @alexjfisher @bastelfreak

Hoping to get this out later today. Let me know if there's anything you see wrong!

File: doc/faq.mkd [hunk 1]
```diff
@@ -1,6 +1,41 @@
 Frequently Asked Questions
 ==========================
 
+### How can run `puppet generate types` for each changed environment on deployment?
```
Comment (alexjfisher) [inline doc/faq.mkd]: This heading doesn't read too well.  Is it just missing an `I`? (`How can I run...`)

File: doc/faq.mkd [hunk 2]
```diff
@@ -1,6 +1,41 @@
...
+
+And configuring r10k such:
+```
+$ tail -2 /etc/puppetlabs/r10k/r10k.yaml
+
+postrun: ["/usr/local/bin/generate-puppet-types.sh", "$modifiedenvs"]
+```
+
+Should provide you with the necessary generate type definitions for
+imporoved environment isolation.
```
Comment (alexjfisher) [inline doc/faq.mkd]: typo.  Should read `improved`.

File: doc/faq.mkd (hunk 1 above)
Comment (justinstoller) [reply doc/faq.mkd]: Good catch. I definitely meant to put the `I` in there. Though I'm not sure that's the best either, do you have any preferences for other ways to format this heading?

**Review by Magisus — APPROVED** [review]

**Comment by Magisus** [comment]
Pending those fixes already pointed out.

**Comment by justinstoller** [comment]
updated

**Pull request merged**